src/bin/platforms/freebsd.ts

```diff
@@ -0,0 +1,312 @@
+        gid: parseInt(gid, 10),
+      };
+    } else {
+      return {
+        uid: os.userInfo().uid,
+        gid: os.userInfo().gid,
+      };
+    }
+  }
+
+  /**
+   * Returns the pid of the process running on the defined port
+   */
+  public getPidOfPort(port: number) {
+    try {
+      return child_process.execSync(`sockstat -P tcp -p ${port} -l -q 2> /dev/null | awk '{print $3}' | head -n 1`).toString('utf8').trim();
+    } catch (e) {
+      return null;
+    }
+  }
+
+  /**
+   * Enables rc service for autostart
+   */
+  private async enableService() {
+    try {
+      child_process.execSync(`sysrc ${this.rcServiceName}_enable="YES" 2> /dev/null`);
+    } catch (e) {
+      this.hbService.logger(`WARNING: failed to run "sysrc ${this.rcServiceName}_enable=\"YES\"`, 'warn');
+    }
+  }
+
+  /**
+   * Disables rc service for autostart
+   */
+  private async disableService() {
+    try {
+      child_process.execSync(`sysrc ${this.rcServiceName}_enable="˜NO" 2> /dev/null`);
+    } catch (e) {
+      this.hbService.logger(`WARNING: failed to run "sysrc ${this.rcServiceName}_enable=\"NO\"`, 'warn');
+    }
+  }
+  
+  /**
+   * Check the command is being run as root and we can detect the user
+   */
+  private checkForRoot() {
+    if (process.getuid() !== 0) {
+      this.hbService.logger('ERROR: This command must be executed using sudo on FreeBSD', 'fail');
+      this.hbService.logger(`EXAMPLE: sudo hb-service ${this.hbService.action}`, 'fail');
+      process.exit(1);
+    }
+    if (!process.env.SUDO_USER && !this.hbService.asUser) {
+      this.hbService.logger('ERROR: Could not detect user. Pass in the user you want to run Homebridge as using the --user flag eg.', 'fail');
+      this.hbService.logger(`EXAMPLE: sudo hb-service ${this.hbService.action} --user your-user`, 'fail');
+      process.exit(1);
+    }
+  }
+
+  /**
+   * Checks the user exists
+   */
+  private async checkUser() {
+    try {
+      // check if user exists
+      child_process.execSync(`id ${this.hbService.asUser} 2> /dev/null`);
+    } catch (e) {
+      // if not create the user
+      child_process.execSync(`pw useradd -q -n ${this.hbService.asUser} -s /usr/sbin/nologin 2> /dev/null`);
+      this.hbService.logger(`Created service user: ${this.hbService.asUser}`, 'info');
+    }
+  }
+
+  // TODO: shutdown doesn't exist in jails. Require jailutils, sysctl security.jail.jailed to check, jkill to restart
+  /**
+   * Allows the homebridge user to shutdown and restart the server from the UI
+   * There is no need for full sudo access when running using hb-service
+   */
+  private setupSudo() {
+    try {
+      const npmPath = child_process.execSync('which npm').toString('utf8').trim();
+      const shutdownPath = child_process.execSync('which shutdown').toString('utf8').trim();
+      const sudoersEntry = `${this.hbService.asUser}    ALL=(ALL) NOPASSWD:SETENV: ${shutdownPath}, ${npmPath}, /usr/local/bin/npm, /usr/sbin/service, /usr/bin/tail`;
```

Comment:
Do we need /usr/sbin/service, /usr/bin/tail ?

Reply:
I think I'd prefer to just not have "shutdown" / "restart" host option available on FreeBSD, new installs on Debian based systems no longer have this option (apart from the Homebridge Raspberry Pi Image where Homebridge is the main thing).

Reply:
I thought we'd use service to start/stop/restart. tail allows file-based log viewing and was needed before the change to how the rc service is run.

Fine to ditch shutdown. Just to be clear, restarting homebridge is still needed, but restarting the machine is not.